File: backend/app/schemas/predictions.py
# backend/app/schemas/predictions.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict
from datetime import date, datetime


class PredictionRequest(BaseModel):
    espece: str
    race: Optional[str] = None
    animal_id: Optional[int] = None
    lot_id: Optional[int] = None
    prediction_type: str  # growth, production, health, cashflow, compost, water_quality
    horizon_jours: int = Field(30, ge=1, le=365)


class PredictionResponse(BaseModel):
    prediction_id: str
    espece: str
    prediction_type: str
    horizon_jours: int
    predictions: Dict
    confidence: float = Field(..., ge=0, le=100)
    confidence_level: str  # faible, moyenne, élevée
    warnings: List[str] = []
    generated_at: datetime


class GrowthPredictionResponse(BaseModel):
    animal_id: int
    age_actuel_jours: int
    poids_actuel_kg: float
    poids_prevu_jours: List[Dict]  # [{"jour": 30, "poids_min": 10, "poids_max": 15, "poids_moyen": 12.5}]
    date_atteinte_poids_vente: Optional[date]
    retard_croissance_detecte: bool
    recommandations: List[str]


class ProductionPredictionResponse(BaseModel):
    espece: str
    type_production: str  # lait, oeufs, larves
    production_quotidienne_actuelle: float
    production_prevue_15j: float
    production_prevue_30j: float
    production_prevue_90j: float
    saisonnalite_impact: Optional[float]
    recommandations: List[str]


class CashflowPredictionResponse(BaseModel):
    tresorerie_actuelle: float
    entrees_prevues_30j: float
    sorties_prevues_30j: float
    tresorerie_prevue_30j: float
    tresorerie_prevue_60j: float
    tresorerie_prevue_90j: float
    seuil_alerte_atteint: bool
    recommandations: List[str]

File: frontend/src/lib/types/predictions.ts
// lib/types/predictions.ts (complété)
export interface PredictionRequest {
    espece: string;
    race?: string;
    animal_id?: number;
    lot_id?: number;
    prediction_type: 'growth' | 'production' | 'health' | 'cashflow' | 'compost' | 'water_quality';
    horizon_jours: number;
}

export interface PredictionResponse {
    prediction_id: string;
    espece: string;
    prediction_type: string;
    horizon_jours: number;
    predictions: Record<string, any>;
    confidence: number;
    confidence_level: string;
    warnings: string[];
    generated_at: string;
}

export interface GrowthPredictionResponse {
    animal_id: number;
    age_actuel_jours: number;
    poids_actuel_kg: number;
    poids_prevu_jours: Array<{
        jour: number;
        poids_min: number;
        poids_max: number;
        poids_moyen: number;
    }>;
    date_atteinte_poids_vente?: string;
    retard_croissance_detecte: boolean;
    recommandations: string[];
}

export interface ProductionPredictionResponse {
    espece: string;
    type_production: string;
    production_quotidienne_actuelle: number;
    production_prevue_15j: number;
    production_prevue_30j: number;
    production_prevue_90j: number;
    saisonnalite_impact?: number;
    recommandations: string[];
}

export interface CashflowPredictionResponse {
    tresorerie_actuelle: number;
    entrees_prevues_30j: number;
    sorties_prevues_30j: number;
    tresorerie_prevue_30j: number;
    tresorerie_prevue_60j: number;
    tresorerie_prevue_90j: number;
    seuil_alerte_atteint: boolean;
    recommandations: string[];
}